### PR TITLE
Epic login component validate navigation

### PR DESCRIPTION
**Pull request by lmiotti**
### Summary
Sign up fragment was created. Terms and conditions linked to Wolox website. Some fixes were done when calling to another activity.

### Screenshot
image

### Trello Card
https://trello.com/c/qJsqeb7I/20-u1-frontend-login-navegaci%C3%B3n

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/training/ui/RootActivity.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    protected void init() {
+        SharedPreferences sharedPref = this.getSharedPreferences(USER,Context.MODE_PRIVATE);
+        String userKey = sharedPref.getString(USER,DefaultUSER);
+        if (userKey.equals(DefaultUSER)) {
+            // Call LogInActivity
+            Intent logInIntent = new Intent(this, LogInActivity.class);
+            logInIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/RootActivity.java]: You can write more than one flag like this too 
intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    @OnClick(R.id.fragment_login_login_button)
+        public void LogInValidate(){
+        mEmailTxt.setError(null);
+        mPasswordTxt.setError(null);
+            if (mEmailTxt.length() != 0 && mPasswordTxt.length() != 0) {
+                if (!android.util.Patterns.EMAIL_ADDRESS.matcher(mEmailTxt.getText()).matches()) {
+                    mEmailTxt.setError("Please insert a valid email address");
+                } else {
+                    //getPresenter().doLogin(mEmailTxt,mPasswordTxt);
+                    SharedPreferences sharedPref = getActivity().getSharedPreferences(USER, Context.MODE_PRIVATE);
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java]: We should send this to the presenter

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @OnClick(R.id.fragment_login_login_button)
+        public void LogInValidate(){
+        mEmailTxt.setError(null);
+        mPasswordTxt.setError(null);
+            if (mEmailTxt.length() != 0 && mPasswordTxt.length() != 0) {
+                if (!android.util.Patterns.EMAIL_ADDRESS.matcher(mEmailTxt.getText()).matches()) {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java]: Try to make new methods for this like  !isValidEmail(mailText) . You nerver know when the implementation could change :3

File: app/src/main/res/layout/fragment_home.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:text="HOME"/>
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_home.xml]: Get used to set strings in Strings.xml. Just to get a good muscle memory :muscle: :stuck_out_tongue:

File: app/src/main/res/layout/fragment_login.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    <EditText
+        android:id="@+id/fragment_login_email_edit_text"
+        style="@style/EditTextTheme"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/login_email_edit_text_marginTop"
+        android:layout_marginLeft="@dimen/login_edit_text_marginLeft"
+        android:layout_marginRight="@dimen/login_edit_text_marginRight"
+        android:gravity="left"
+        android:textSize="18sp"
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Hardcode dimen . Remember to use dimens.xml

File: app/src/main/res/layout/fragment_login.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    <EditText
+        android:id="@+id/fragment_login_password_edit_text"
+        style="@style/EditTextTheme"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="@dimen/login_edit_text_marginLeft"
+        android:layout_marginRight="@dimen/login_edit_text_marginRight"
+        android:gravity="left"
+        android:textSize="18sp"
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Same here and in the next 3 items

File: app/src/main/res/layout/fragment_signup.xml [hunk 7]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_signup.xml:1]: Check Hardcode dimens in all this file please :smiley:

**Review by DemianIac — CHANGES_REQUESTED** [review]
Some minor changes ! :smiley:

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java (hunk 2 above)
Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/training/ui/login/LogInFragment.java]: Saving key values and calling HomeActivity is sent to Presenter in new version "epic-login-component-validate-navigation-access"

**Comment by lmiotti** [comment]
### Summary
Harcoded dimens solved, HOME string is referenced to Strings.xml, and FLAGS are rewritten in one line.

### Screenshot
image

### Trello Card
https://trello.com/c/qJsqeb7I/20-u1-frontend-login-navegaci%C3%B3n

**Review by DemianIac — APPROVED** [review]
:+1:

**Pull request merged**